src/main/java/me/tomsdevsn/hetznercloud/objects/general/FWApplicationTarget.java

```diff
@@ -0,0 +1,38 @@
+package me.tomsdevsn.hetznercloud.objects.general;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+
+@Data
+@SuperBuilder
+@NoArgsConstructor
+@AllArgsConstructor
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class FWApplicationTarget {
+
+    @JsonProperty("label_selector")
+    private FWLabelSelector labelSelector;
+    private FWServerRef server;
+    private TargetType type;
+
+    public void isValidOrThrow() throws RuntimeException {
+        final boolean isServerType = (getType() == TargetType.server);
+        final boolean isServerNull = (getServer() == null);
+
+        final boolean isLabelSelectorType = (getType() == TargetType.label_selector);
+        final boolean isLabelSelectorNull = (getLabelSelector() == null);
+
+        if (isServerType && (isServerNull || !isLabelSelectorNull))
+            throw new RuntimeException(
+                    "TargetType is \"server\", either \"server\" is null or \"labelSelector\" is not");
+
+        if (isLabelSelectorType && (isLabelSelectorNull || !isServerNull))
+            throw new RuntimeException(
+                    "TargetType is \"label_selector\", either \"labelSelector\" is null or \"server\" is not");
+    }
```

Comment:
This check is not needed.

The API ignores e.g. `label_selector` if `type=server` is set.

```
curl -s -H "Authorization: Bearer ${HCLOUD_TOKEN}" -H "Content-Type: application/json" https://api.hetzner.cloud/v1/firewalls/1/actions/apply_to_resources -X POST -d '{"apply_to": [{"server": {"id": 100}, "label_selector": {"selector": "env=kap0tt"}, "type": "server"}]}'
{
  "actions": [
    {
      "id": 608754500,
      "command": "apply_firewall",
      "status": "running",
      "progress": 0,
      "started": "2022-08-12T12:35:06+00:00",
      "finished": null,
      "resources": [
        {
          "id": 1,
          "type": "server"
        },
        {
          "id": 100,
          "type": "firewall"
        }
      ],
      "error": null
    }
  ]
}
```